paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -188,14 +203,173 @@ def deep_delete(self, kube_client: KubeClient) -> None:
+                    resource=V2beta1ResourceMetricSource(
+                        name="cpu", target_average_utilization=target
+                    ),
+                )
+            )
+        elif metrics_provider == "http":
+            metrics.append(
+                V2beta1MetricSpec(
+                    type="Pods",
+                    pods=V2beta1PodsMetricSource(
+                        metric_name="http", target_average_value=target
+                    ),
+                )
+            )
+        elif metrics_provider == "uwsgi":
+            metrics.append(
+                V2beta1MetricSpec(
+                    type="Pods",
+                    pods=V2beta1PodsMetricSource(
+                        metric_name="uwsgi", target_average_value=target
+                    ),
+                )
+            )
+        else:
+            self.logging.error("Wrong metrics specified")
+            return
+
+        body = V2beta1HorizontalPodAutoscaler(
+            kind="HorizontalPodAutoscaler",
+            metadata=V1ObjectMeta(
+                name=self.item.metadata.name, namespace=self.item.metadata.namespace
+            ),
+            spec=V2beta1HorizontalPodAutoscalerSpec(
+                max_replicas=max_replicas,
+                min_replicas=min_replicas,
+                metrics=metrics,
+                scale_target_ref=V2beta1CrossVersionObjectReference(
+                    kind="Deployment", name=self.item.metadata.name
+                ),
+            ),
+        )
+        self.logging.debug(body)
+        if not hpa_exists:
+            self.logging.info(
+                "Creating new HPA for {metrics_provider} {self.item.metadata.name}/name in {self.item.metadata.namespace}"
+            )
+            kube_client.autoscaling.create_namespaced_horizontal_pod_autoscaler(
+                namespace=self.item.metadata.namespace, body=body, pretty=True
+            )
+        else:
+            self.logging.info(
+                "Updating new HPA for {metrics_provider} {self.item.metadata.name}/name in {self.item.metadata.namespace}/namespace"
+            )
+            kube_client.autoscaling.patch_namespaced_horizontal_pod_autoscaler(
+                name=self.item.metadata.name,
+                namespace=self.item.metadata.namespace,
+                body=body,
+                pretty=True,
+            )
+
+    def exists_hpa(self, kube_client: KubeClient) -> bool:
+        try:
+            kube_client.autoscaling.read_namespaced_horizontal_pod_autoscaler(
+                name=self.item.metadata.name, namespace=self.item.metadata.namespace
+            )
+        except ValueError as e:
+            self.logging.error(
+                "Error occurs for {self.item.metadata.name}/name in {self.item.metadata.namespace}/namespace.  \
+                This could a bug in k8s. For detail: https://github.com/kubernetes-client/python/issues/415"
+            )
+            self.logging.error(e)
```

Comment:
It's better to log an error with all corresponding context, in this case - output `e` within the previous message using the same `logging .error`.